Move all test files in the `js/src/data` folder to test subfolder and separate checking functions of immutable state

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

It's a follow-up of https://github.com/woocommerce/google-listings-and-ads/pull/1150#discussion_r766756148.

In order to avoid confusion between the helper.js file that includes utils functions for testing and the plugin implementation, move all test files in the *js/src/data* folder to the *test* subfolder in it.

- Move ***.test.js** files.
- Move deep freeze and reference checking functions to a separate file.

### Detailed test instructions:

1. `npm run test:js:watch`
2. Check if all test cases are passed.
3. Make changes to state directly or other state tree node references in the *reducer.js* to see if the deep freeze and reference checking functions are able to discover unexpected mutations on state.

### Changelog entry

eason9487 committed:
Move test files in the js/src/data folder to the 'test' sub-folder

eason9487 committed:
Ignore `/__helpers__/` pattern from running jest test and coverage analysis

eason9487 committed:
Move deep freeze and ref check functions to a separate file

puntope commented on js/src/data/test/__helpers__/index.js:
```diff
@@ -0,0 +1,105 @@
+	return Object.freeze( object );
+}
+
+function attachRef( object, ignore, checkingGroups = [], paths = [] ) {
+	const refKey = '__refForCheck';
+
+	for ( const [ name, value ] of Object.entries( object ) ) {
+		if ( isPlainObject( value ) ) {
+			const checkPaths = [ ...paths, name ];
+			attachRef( value, ignore, checkingGroups, checkPaths );
+
+			const currentPath = checkPaths.join( '.' );
+			if ( ! ignore.includes( currentPath ) ) {
+				const ref = {
+					whyFail: `If there's no mutation at \`state.${ currentPath }\`, it should be kept the same reference.`,
+				};
+				checkingGroups.push( {
+					ref,
+					refPath: `${ currentPath }.${ refKey }`,
+				} );
+				value[ refKey ] = ref;
+			}
+		}
+	}
+	return checkingGroups;
+}
```
💅 
Missing docs

eason9487 replied:
Added by https://github.com/woocommerce/google-listings-and-ads/pull/1153/commits/9ca8505925788cd3b588e18df2efec8c65a8f748

puntope reviewed (approved):
✅  LGTM Nice work!

- Test passed ✅ 
- Mutating the state creates error  `TypeError: Cannot assign to read only property 'rates' of object '#<Object>` ✅ 
- Visual testing of the files looks fine ✅ 

💅  

Added a comment regarding a missing docs in `attachRef`

eason9487 committed:
Add JSDoc for attachRef function in the testing heler.

Address https://github.com/woocommerce/google-listings-and-ads/pull/1153#discussion_r769548200

eason9487 committed:
Fix typo

eason9487 committed:
Rephrase the test description for the MC issues reducer.

Address https://github.com/woocommerce/google-listings-and-ads/pull/1150#discussion_r768862820

Since it only changes the test description,
I directly update the description here to avoid code conflict
and having to redo the same thing again before merging.
